### ISSUE TITLE
Annotations in function parameters incorrect

### ISSUE THREAD

**Issue by mariotaku** [krzysztofzablocki/Sourcery#283]
# Version
0.6.0

Annotations in function parameters incorrect, seems it contains annotations from other functions. For example, `functionB(/* sourcery:annotation=param3 */param3: String)` should have annotation `["annotation": param3]`, however, it has other values instead.

# POC

## Template file

````
// sourcery:file:bug_output.txt
<%_ for type in types.protocols where type.annotations["testProtocol"] != nil { -%>
    <%_ for method in type.methods where method.annotations["testFunction"] != nil { -%>
<%= method.name%>: <%= method.parameters %>
    <%_ } -%>
<%_ } -%>
// sourcery:end
````

## Source file

````swift
// sourcery: testProtocol
protocol TestProtocol {
    // sourcery: testFunction
    func functionA()
    // sourcery: testFunction
    func functionB(/* sourcery:annotation=param1 */param1: Int, /* sourcery:annotation=param2 */param2: Int)
    // sourcery: testFunction
    func functionB(/* sourcery:annotation=param3 */param3: String)
}
````

## Output

````
functionA(): []
functionB(param1: Int, param2: Int): [MethodParameter: argumentLabel = Optional("param1"), name = param1, typeName = Int, typeAttributes = [:], defaultValue = nil, annotations = ["annotation": param1], MethodParameter: argumentLabel = Optional("param2"), name = param2, typeName = Int, typeAttributes = [:], defaultValue = nil, annotations = ["annotation": param2]]
functionB(param3: String): [MethodParameter: argumentLabel = Optional("param3"), name = param3, typeName = String, typeAttributes = [:], defaultValue = nil, annotations = ["annotation": param2, "testFunction": 1]]
````

**Comment by ilyapuchka** [comment]
Hey, @mariotaku ! Thanks for reporting. I think it's related to #282 , I will add these to the test cases to the suite in the pr.

**Comment by ilyapuchka** [comment]
Should be resolved with #282
